crates/risc0-bonsai/src/host.rs

```diff
@@ -405,11 +405,16 @@ impl<'host> Zkvm for Risc0BonsaiHost<'host> {
 
     type Error = anyhow::Error;
 
-    fn verify<'a>(
-        _serialized_proof: &'a [u8],
-        _code_commitment: &Self::CodeCommitment,
-    ) -> Result<&'a [u8], Self::Error> {
-        unimplemented!();
+    fn verify(
+        serialized_proof: &[u8],
+        code_commitment: &Self::CodeCommitment,
+    ) -> Result<Vec<u8>, Self::Error> {
+        let receipt: Receipt = bincode::deserialize(serialized_proof)?;
+
+        #[allow(clippy::clone_on_copy)]
```

Comment:
What's that?

Reply:
CodeCommitment type is copyable (inner is [u32;8]), but we are calling clone on it. Hence it is unnecessary and can be just dereferenced with *. It was like that previously hence didn't give much attention just copy-pasted.